### PR TITLE
Added Custom 404 Error Page

### PR DESCRIPTION
**Pull request by spider-sauce**
### Description

Added custom 404 page.
### Motivation and Context

Replace the default Docusaurus 404 component.
### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation fix (typos, incorrect content, missing content, etc.)

### Checklist

- [x] I have read the [contributing](https://github.com/saucelabs/sauce-docs/blob/main/docs/contributing.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by spider-sauce** [comment]
@sweeneyskirt-sl and / or @kimsauce could you take a look at this and let me know what you think? I did some basic styling but if you have suggestions let me know. BTW you have to test via `npm run build` + `npm run serve`, it won't work on the client side with `yarn start || npm run start`

File: static/404.html [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    <link data-react-helmet="true" rel="shortcut icon" href="/img/favicon.ico">
+</head>
+<body>
+    <div class="container">
+        <div class="flex header"><h2>404 Error | Sauce Docs: Content Does Not Exist!</h2></div>
+        <div class="flex sauce_bot">
+            <img src="/img/Manual_Bot_EXPORT.gif">
+        </div>
+        <div class="flex message">
+                <p>Oops, it looks like the page you're looking for either no longer exists, or moved.</p>
```
Comment (sweeneyskirt-sl) [inline static/404.html:17]: You actually don't need a comma here.

File: static/404.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+<head>
+    <title>
+        404 Error | Sauce Docs
+    </title>
+    <link rel="stylesheet" href="/css/404.css">
+    <link data-react-helmet="true" rel="shortcut icon" href="/img/favicon.ico">
+</head>
+<body>
+    <div class="container">
+        <div class="flex header"><h2>404 Error | Sauce Docs: Content Does Not Exist!</h2></div>
```
Comment (sweeneyskirt-sl) [inline static/404.html:12]: Do we want to say "content does not exist"? That seems a little bleak. Maybe something like "We can't find that page" or even "that URL does not exist"? 

Sorry to be such a word-nerd. I just would like to believe that _the content_ probably does exist somewhere, just not at the URL you've entered. :)

**Review by sweeneyskirt-sl — COMMENTED** [review]
LOVE LOVE LOVE the bot! I especially like how the little antenna bounces at the end! I could watch it ll day. LOL

Couple comments on the text, but overall looks good.

File: static/404.html (hunk 2 above)
Comment (spider-sauce) [reply static/404.html:12]:
```suggestion
        <div class="flex header"><h2>404 Error | Sauce Docs: URL does not exist</h2></div>
```

File: static/404.html (hunk 1 above)
Comment (spider-sauce) [reply static/404.html:17]:
```suggestion
                <p>Oops, it looks like the page you're looking for either no longer exists or moved.</p>
```

**Comment by spider-sauce** [comment]
> Hi @spider-sauce , it worked perfectly and love the gif!
> 
> The only thing I might change is maybe shrinking the image a little bit so that you can see the text below it. I had to zoom out in my browser to 75% in order to see everything.

@kimsauce, I shrunk the image proportionally (50%), if you want to check again. Let me know if it still looks funky.

**Review by sweeneyskirt-sl — APPROVED** [review]
righteous. :shipit:

**Comment by spider-sauce** [comment]
> righteous. :shipit:

@sweeneyskirt-sl , do you think we'll get pushback on accessibility because it's a .gif and not a static image? On one hand it's an error page, so maybe it's not as crucial? 🤷

File: static/404.html [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    <link data-react-helmet="true" rel="shortcut icon" href="/img/favicon.ico">
+</head>
+<body>
+    <div class="container">
+        <div class="flex header"><h2>404 Error | Sauce Docs: URL does not exist</h2></div>
+        <div class="flex sauce_bot">
+            <img class="gif" src="/img/Manual_Bot_EXPORT.gif" alt="Sauce Bot Animated GIF">
+        </div>
+        <div class="flex message">
+                <p>Oops, it looks like the page you're looking for either no longer exists or moved.</p>
```
Comment (kimsauce) [inline static/404.html:17]: @spider-sauce sorry, one more thing! Would you be ok with change? 
_Oops! It looks like the page you're looking for no longer exists or has been moved._

**Pull request merged**